### ISSUE TITLE
KServe WebHook throws 500 due to loading root certificates

### ISSUE THREAD

**Issue by kimwnasptd** [kubeflow/manifests#2553]
@DnPlas bumped into this in https://github.com/kubeflow/manifests/pull/2552
https://github.com/kubeflow/manifests/actions/runs/6535263237/job/17747221819

This could be a transient issue that would require us to wait before applying the rest of the manifests, or bluntly reapplying. Will try to reproduce locally to see if it's indeed a race condition

cc @yuzisun 

```
certificate.cert-manager.io/serving-cert unchanged
issuer.cert-manager.io/selfsigned-issuer unchanged
mutatingwebhookconfiguration.admissionregistration.k8s.io/inferenceservice.serving.kserve.io configured
validatingwebhookconfiguration.admissionregistration.k8s.io/clusterservingruntime.serving.kserve.io configured
validatingwebhookconfiguration.admissionregistration.k8s.io/inferencegraph.serving.kserve.io configured
validatingwebhookconfiguration.admissionregistration.k8s.io/inferenceservice.serving.kserve.io configured
validatingwebhookconfiguration.admissionregistration.k8s.io/servingruntime.serving.kserve.io configured
validatingwebhookconfiguration.admissionregistration.k8s.io/trainedmodel.serving.kserve.io configured
Error from server (InternalError): error when creating "STDIN": Internal error occurred: failed calling webhook "clusterservingruntime.kserve-webhook-server.validator": could not get REST client: unable to load root certificates: unable to parse bytes as PEM block
Error from server (InternalError): error when creating "STDIN": Internal error occurred: failed calling webhook "clusterservingruntime.kserve-webhook-server.validator": could not get REST client: unable to load root certificates: unable to parse bytes as PEM block
Error from server (InternalError): error when creating "STDIN": Internal error occurred: failed calling webhook "clusterservingruntime.kserve-webhook-server.validator": could not get REST client: unable to load root certificates: unable to parse bytes as PEM block
Error from server (InternalError): error when creating "STDIN": Internal error occurred: failed calling webhook "clusterservingruntime.kserve-webhook-server.validator": could not get REST client: unable to load root certificates: unable to parse bytes as PEM block
Error from server (InternalError): error when creating "STDIN": Internal error occurred: failed calling webhook "clusterservingruntime.kserve-webhook-server.validator": could not get REST client: unable to load root certificates: unable to parse bytes as PEM block
Error from server (InternalError): error when creating "STDIN": Internal error occurred: failed calling webhook "clusterservingruntime.kserve-webhook-server.validator": could not get REST client: unable to load root certificates: unable to parse bytes as PEM block
Error from server (InternalError): error when creating "STDIN": Internal error occurred: failed calling webhook "clusterservingruntime.kserve-webhook-server.validator": could not get REST client: unable to load root certificates: unable to parse bytes as PEM block
Error from server (InternalError): error when creating "STDIN": Internal error occurred: failed calling webhook "clusterservingruntime.kserve-webhook-server.validator": could not get REST client: unable to load root certificates: unable to parse bytes as PEM block
Error from server (InternalError): error when creating "STDIN": Internal error occurred: failed calling webhook "clusterservingruntime.kserve-webhook-server.validator": could not get REST client: unable to load root certificates: unable to parse bytes as PEM block
```

**Comment by kimwnasptd** [comment]
I tried to re-run this action but looks like it was re-triggered

**Comment by kimwnasptd** [comment]
@yuzisun @DnPlas I found the bug. In `v0.11.1` I see two of the webhooks that reference an incorrect Cert Manager `Certificate`:
1. The `clusterservingruntime.serving.kserve.io` ValidationWebhookConfiguration https://github.com/kserve/kserve/blob/eea3ae94bd5db268b7a01b2bec87cdd5dd7d668f/install/v0.11.1/kserve_kubeflow.yaml#L21371
2. The `servingruntime.serving.kserve.io` ValidationWebhookConfiguration https://github.com/kserve/kserve/blob/eea3ae94bd5db268b7a01b2bec87cdd5dd7d668f/install/v0.11.1/kserve_kubeflow.yaml#L21467

Those two should instead have an annotation `cert-manager.io/inject-ca-from: kubeflow/serving-cert` like the rest of the webhooks. After making this change the installation continues as expected. This will mean we'll need a new minor version of KServe for this.

**Comment by kimwnasptd** [comment]
In `v0.11.0` those were had the correct values. Maybe the culprit is https://github.com/kserve/kserve/pull/3083?

**Comment by DnPlas** [comment]
Just FYI, the Kserve WG is working on a fix atm: https://github.com/kserve/kserve/pull/3188.

**Comment by umka1332** [comment]
> Just FYI, the Kserve WG is working on a fix atm: [kserve/kserve#3188](https://github.com/kserve/kserve/pull/3188).

@kimwnasptd fyi: [KServe 0.11.2 release](https://github.com/kserve/kserve/releases/tag/v0.11.2) was released with this fix

**Comment by juliusvonkohout** [comment]
This should be closed by https://github.com/kubeflow/manifests/pull/2627

/close
